### ISSUE TITLE
Sound variations by pitch

### ISSUE THREAD

**Issue by tacotexmex** [minetest/minetest_game#2085]
By applying a slight pitch to sounds, footsteps and dig sounds for example, we may instantly offer more variations of repeating sounds. This is desirable in most games, in order to not make the tasks sound as repetitive.

In this example the pitch is applied as either 0.9, 1.0 or 1.1, thus the player hear 3 variations of each sound if applied on every `sound_play`. 

```
math.randomseed(os.time())
function vpitch()
	return (100 - (math.random(-1, 1)) * 10) / 100
end
```
☝️😊 _Sorry for my crude math!_

Looking in `default.node_sound_defaults` I can see that the sound definitions comes predefined so that would have to be modified if this is to be implemented.

**Comment by SmallJoker** [comment]
1) Wrong repository -> Game engine/API requests belong here: https://github.com/minetest/minetest/issues
2) This feature already exists since last June: https://github.com/minetest/minetest/pull/5960
3) The PR was merged 7 days after the 0.4.16 release, which means you'll have to include that patch manually into your 0.4.16 stable, get a very early build after the release or get any 0.5.0-dev build.

**Comment by SmallJoker** [comment]
Apparently I've read this issue too quickly. I took this as a feature request in the wrong repository. Of course, some additional variety in the sounds would be nice to have :)

**Comment by tacotexmex** [comment]
Your swiftness laudable but yes, I think I'm in the right place and legitimate. 😁

**Comment by jastevenson303** [comment]
The footstep sound is played in the engine.

**Comment by paramat** [comment]
Is there actually a way to alter pitch of footstep sounds that are triggered in the engine, from the game? Have you tried it? Idea is fone though.

**Comment by paramat** [comment]
```
minetest.register_node("aaa:aaa", {
    sounds = {
        footstep = {name = "default_place_node", gain = 1.0, pitch = 1.5}
    }
})
```
Doesn't seem so, this or a random value would always play the same pitch each time.

**Comment by alexdevteam** [comment]
The pitch value used in the registation function is used through all the gameplay. So, if you input a random pitch value in the registation function, and that random value lands on "1.2" for example, that value and ONLY that value will be used as the pitch for the sound selected.

**Comment by paramat** [comment]
Needs a change to the sound API, so an engine issue should be opened if it hasn't already.

**Comment by HybridDog** [comment]
There is sound pitch in minetest.

**Comment by jastevenson303** [comment]
U can change the pitch of node sounds, but only once.

**Comment by paramat** [comment]
Engine issue.

**Comment by alexdevteam** [comment]
This is STILL going? Holy crap
